driver-embedded/src/main/com/mongodb/embedded/client/MongoDBCAPIHelper.java

```diff
@@ -165,4 +172,49 @@ private static void checkInitialized() {
 
     private MongoDBCAPIHelper() {
     }
+
+    /**
+     * Represents libmongodbcapi_init_params
+     */
+    public static class MongoDBCAPIInitParams extends Structure implements MongoDBCAPI.InitParams {
+        // CHECKSTYLE.OFF: VisibilityModifier
+        public String yamlConfig;
+        public long logFlags;
+        public Callback logCallback;
+        public String userData;
+        // CHECKSTYLE.ON: VisibilityModifier
+
+        MongoDBCAPIInitParams(final MongoEmbeddedSettings settings) {
+            super();
+            this.yamlConfig = settings.getYamlConfig();
+            this.logFlags = settings.getLogLevel().getLevel();
+            this.logCallback = settings.getLogLevel() == MongoEmbeddedSettings.LogLevel.LOGGER ? new LogCallback() : null;
+            this.userData = settings.getApplicationData();
+        }
+
+        protected List<String> getFieldOrder() {
+            return asList("yamlConfig", "logFlags", "logCallback", "userData");
+        }
+    }
+
+    static class LogCallback implements Callback {
```

Comment:
Represents `libmongodbcapi_log_callback`.

```
typedef void (*libmongodbcapi_log_callback)(
    void* user_data, const char* message, const char* component, const char* context, int severity);
```

Logs the messages via the standard logging infrastructure.  Maps the server log levels to the Java levels and the logs are formatted in the same style as the mongod.log.

In theory we could allow users to provide a custom `LogCallback` but for the time being its locked down and this is the only implementation.